src/decide.ts

```diff
@@ -227,84 +267,89 @@ function releaseGroupDoesMatch(
+			return { decision: Decision.RESOLUTION_MISMATCH };
+		}
+	}
+	if (!link) {
+		return { decision: Decision.NO_DOWNLOAD_LINK };
+	}
+
+	if (findBlockedStringInReleaseMaybe(searchee, blockList)) {
+		return { decision: Decision.BLOCKED_RELEASE };
+	}
 
+	const result = await snatch(candidate);
 	if (result.isErr()) {
-		return result.unwrapErr() === SnatchError.RATE_LIMITED
+		const err = result.unwrapErr();
+		return err === SnatchError.RATE_LIMITED
 			? { decision: Decision.RATE_LIMITED }
-			: { decision: Decision.DOWNLOAD_FAILED };
+			: err === SnatchError.MAGNET_LINK
+				? { decision: Decision.MAGNET_LINK }
+				: { decision: Decision.DOWNLOAD_FAILED };
 	}
+	const metafile = result.unwrap();
+	cacheTorrentFile(metafile);
+	candidate.size = metafile.length; // Trackers can be wrong
 
-	const candidateMeta = result.unwrap();
-
-	if (hashesToExclude.includes(candidateMeta.infoHash)) {
-		return { decision: Decision.INFO_HASH_ALREADY_EXISTS };
+	if (hashesToExclude.includes(metafile.infoHash)) {
+		return { decision: Decision.INFO_HASH_ALREADY_EXISTS, metafile };
 	}
-	const sizeMatch = compareFileTreesIgnoringNames(candidateMeta, searchee);
-	const perfectMatch = compareFileTrees(candidateMeta, searchee);
 
+	const perfectMatch = compareFileTrees(metafile, searchee);
 	if (perfectMatch) {
-		return { decision: Decision.MATCH, metafile: candidateMeta };
+		return { decision: Decision.MATCH, metafile };
 	}
-	if (
-		sizeMatch &&
-		matchMode !== MatchMode.SAFE &&
-		searchee.files.length === 1
-	) {
-		return {
-			decision: Decision.MATCH_SIZE_ONLY,
-			metafile: candidateMeta,
-		};
+
+	const sizeMatch = compareFileTreesIgnoringNames(metafile, searchee);
+	if (sizeMatch && matchMode !== MatchMode.SAFE) {
+		return { decision: Decision.MATCH_SIZE_ONLY, metafile };
 	}
 
 	if (matchMode === MatchMode.PARTIAL) {
-		const partialSizeMatch = compareFileTreesPartialIgnoringNames(
-			candidateMeta,
-			searchee,
-		);
+		const partialSizeMatch = comparePartialSizeOnly(metafile, searchee);
 		if (!partialSizeMatch) {
-			return { decision: Decision.SIZE_MISMATCH };
+			return { decision: Decision.PARTIAL_SIZE_MISMATCH, metafile };
 		}
-
-		const partialMatch = compareFileTreesPartial(candidateMeta, searchee);
+		const partialMatch = compareFileTreesPartial(metafile, searchee);
 		if (partialMatch) {
-			return {
-				decision: Decision.MATCH_PARTIAL,
-				metafile: candidateMeta,
-			};
+			return { decision: Decision.MATCH_PARTIAL, metafile };
 		}
 	} else if (!sizeMatch) {
-		return { decision: Decision.SIZE_MISMATCH };
+		return { decision: Decision.SIZE_MISMATCH, metafile };
 	}
 
-	return { decision: Decision.FILE_TREE_MISMATCH };
+	return { decision: Decision.FILE_TREE_MISMATCH, metafile };
 }
 
-function existsInTorrentCache(infoHash: string): boolean {
-	return existsSync(
-		path.join(appDir(), TORRENT_CACHE_FOLDER, `${infoHash}.cached.torrent`),
+function existsInTorrentCache(infoHash: string | undefined): boolean {
```

Comment:
I think this function should take `string` rather than `string | undefined`. Let's push the null checks to the callers